csf_tz/csftz_hooks/purchase_order.py

```diff
@@ -0,0 +1,39 @@
+import frappe
+from frappe import _
+
+
+@frappe.whitelist()
+def update_po_status(status, name):
+    from erpnext.buying.doctype.purchase_order.purchase_order import update_status
+
+
+    if status != "Submitted":
+        update_status(status, name)
+        return
+
+    csf_tz_settings = frappe.get_doc("CSF TZ Settings")
+    if csf_tz_settings.allow_reopen_of_po_based_on_role == 1:
+        roles = frappe.get_roles()
+        if csf_tz_settings.role in roles:
```

Comment:
it won't throw error since if the value is not set means it will None, there it will check if None is in roles and if not it will execute code from else condition.